servers/physics_3d/godot_shape_3d.cpp

```diff
@@ -847,41 +847,67 @@ void GodotConvexPolygonShape3D::project_range(const Vector3 &p_normal, const Tra
 
 	const Vector3 *vrts = &mesh.vertices[0];
 
-	for (int i = 0; i < vertex_count; i++) {
-		real_t d = p_normal.dot(p_transform.xform(vrts[i]));
+	if (vertex_count > 3*extreme_vertices.size()) {
+		// For a large mesh, two calls to get_support() is faster than a full
+		// scan over all vertices.
 
-		if (i == 0 || d > r_max) {
-			r_max = d;
-		}
-		if (i == 0 || d < r_min) {
-			r_min = d;
+		Vector3 n = p_transform.basis.xform_inv(p_normal);
```

Comment:
using xform_inv will result in scale no longer working. May need to  normalized it (so at least for uniform scale works), but I think this should be handled by eventually passing the both transform and the actual inverse.

Reply:
I didn't realize Basis could include scale factors.  Is the correct behavior to just call `normalized()` on the result?  That should be fine for uniform scaling.  It isn't obvious to me what the intended behavior would be if you give it a Basis with different scale factors along the axes.